ui/components/app/wallet-overview/aggregated-percentage-overview.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { zeroAddress } from 'ethereumjs-util';
+import {
+  getCurrentCurrency,
+  getSelectedAccount,
+  getShouldHideZeroBalanceTokens,
+  getTokensMarketData,
+} from '../../../selectors';
+
+import { useAccountTotalFiatBalance } from '../../../hooks/useAccountTotalFiatBalance';
+import { isEqualCaseInsensitive } from '../../../../shared/modules/string-utils';
+import { formatValue, isValidAmount } from '../../../../app/scripts/lib/util';
+import { getIntlLocale } from '../../../ducks/locale/locale';
+import {
+  Display,
+  TextColor,
+  TextVariant,
+  FontWeight,
+} from '../../../helpers/constants/design-system';
+import { Box, Text } from '../../component-library';
+
+export const AggregatedPercentageOverview = () => {
+  const tokensMarketData = useSelector(getTokensMarketData);
+  const locale = useSelector(getIntlLocale);
+  const fiatCurrency = useSelector(getCurrentCurrency);
+  const selectedAccount = useSelector(getSelectedAccount);
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+  // Get total balance (native + tokens)
+  const { totalFiatBalance, orderedTokenList } = useAccountTotalFiatBalance(
+    selectedAccount,
+    shouldHideZeroBalanceTokens,
+  );
+
+  let totalFiat1dAgo = 0;
+  orderedTokenList.forEach((token) => {
+    if (token.address) {
+      // This is a regular ERC20 token
+      // find the relevant pricePercentChange1d in tokensMarketData
+      // Iterate through the keys of the record
+      let found;
+      for (const key in tokensMarketData) {
+        if (
+          isEqualCaseInsensitive(
+            tokensMarketData[key].tokenAddress,
+            token.address,
+          )
+        ) {
+          found = tokensMarketData[key];
+        }
+      }
+
+      const tokenFiat1dAgo =
+        found?.pricePercentChange1d !== undefined && token?.fiatBalance
+          ? token?.fiatBalance / (1 + found?.pricePercentChange1d / 100)
+          : token?.fiatBalance ?? 0;
+
+      totalFiat1dAgo += Number(tokenFiat1dAgo);
+    } else {
+      // native token
+      const nativePricePercentChange1d =
+        tokensMarketData?.[zeroAddress()]?.pricePercentChange1d;
+
+      const nativeFiat1dAgo =
+        nativePricePercentChange1d !== undefined && token?.fiatBalance
+          ? token.fiatBalance / (1 + nativePricePercentChange1d / 100)
+          : token?.fiatBalance ?? 0;
+
+      totalFiat1dAgo += Number(nativeFiat1dAgo);
+    }
+  });
+
+  const totalBalance: number = Number(totalFiatBalance);
+  const totalBalance1dAgo = totalFiat1dAgo;
+
+  const amountChange = totalBalance - totalBalance1dAgo;
+  const percentageChange = (amountChange / totalBalance1dAgo) * 100 || 0;
```

Comment:
Do you anticipate that this logic will be reused by other components? It doesn't seem like it for now.

Reply:
Hmmm im not anticipating for this to be used elsewhere;

Reply:
I know the logic isn't to heavy but maybe for some of these calculations we should break them into functions in a helper file so that they are easy to add tests for.

Reply:
Thanks guys; i pushed a refactor logic [0336501](https://github.com/MetaMask/metamask-extension/pull/27108/commits/0336501f3008e40551b5837721dd9302132daaae) where i extracted the calculation into its own helper fct; lemme know anything 🙏